### PR TITLE
BUG: TypeError in groupby pct_change when fill_method is None

### PR DESCRIPTION
**Pull request by fujiaxiang**
- [x] xref one of two issues in #30463
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -869,6 +869,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrame.groupby` when using axis=1 and having a single level columns index (:issue:`30208`)
 - Bug in :meth:`DataFrame.groupby` when using nunique on axis=1 (:issue:`30253`)
 - Bug in :meth:`GroupBy.quantile` with multiple list-like q value and integer column names (:issue:`30289`)
+- Bug in :meth:`GroupBY.pct_change` and `SeriesGroupBy.pct_change` causes TypeError when fill_method is None (GH30463)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: can you also add ``:meth:`` for Series, use double-backticks on TypeError, fill_method, None

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. small comments, ping on green.

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (fujiaxiang) [reply doc/source/whatsnew/v1.0.0.rst]: done

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -869,6 +869,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrame.groupby` when using axis=1 and having a single level columns index (:issue:`30208`)
 - Bug in :meth:`DataFrame.groupby` when using nunique on axis=1 (:issue:`30253`)
 - Bug in :meth:`GroupBy.quantile` with multiple list-like q value and integer column names (:issue:`30289`)
+- Bug in :meth:`GroupBy.pct_change` and :meth:`SeriesGroupBy.pct_change` causes ``TypeError`` when ``fill_method`` is ``None`` (GH30463)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: can you use the issue format (see other notes), e.g. ``:issue:``, otherwise they won't render

**Review by jreback — CHANGES_REQUESTED** [review]
doc comment, ping on green.

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (fujiaxiang) [reply doc/source/whatsnew/v1.0.0.rst]: That was such a stupid mistake. Sorry for the carelessness.
Fixed now.

**Review by TomAugspurger — APPROVED** [review]
FYI, doc references need to be to pandas.core.groupby. Will fix those in a followup.

Thanks!

**Pull request merged**